### PR TITLE
(DOCSP-20436): Add Null Support for Optional Types in Realm Schema

### PR DESCRIPTION
**Pull request by mongodben**
## Pull Request Info

Document how to add null support for optional types in a realm schema

### Jira

- https://jira.mongodb.org/browse/DOCSP-20436

### Staged Changes (Requires MongoDB Corp SSO)

- [Enforce a Schema page > Validate Null Types section](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCSP-20436/schemas/enforce-a-schema#validate-null-types)

### Review Guidelines

[REVIEWING.md](https://github.com/mongodb/docs-realm/blob/master/REVIEWING.md)

### CODE REVIEW

File: source/schemas/enforce-a-schema.txt [hunk 1]
```diff
@@ -43,3 +43,20 @@ Procedure
...
       
       .. include:: /includes/steps/enforce-a-schema-cli.rst
+
+.. _validate-null-types:
+
+Validate Null Types
+-------------------
+
+You can configure {+service-short+} to pass schema validation for any ``null`` values
+when their field types are set as optional in the schema.
```
Comment (sumedhamehta) [inline source/schemas/enforce-a-schema.txt:53]: Can we write something about why this isn't the default behavior?

maybe "_Even though null is a separate type in MongoDB's bson types <link to bson types>"_

Comment (mongodben) [reply source/schemas/enforce-a-schema.txt:53]: this could be a note perhaps. will add something

**Review by sumedhamehta — APPROVED** [review]

File: source/schemas/enforce-a-schema.txt [hunk 2]
```diff
@@ -43,3 +43,28 @@ Procedure
...
+Enabling null type validation allows the value for a field to be persisted as
+the type in the schema or the :manual:`BSON null </reference/bson-types/>` type.
+If you do not enable null type schema validation, {+service-short+} rejects ``null``
+values passed to optional fields. 
+
+.. note:: 
+
+   Schema fields are not nullable by default because ``null`` is a unique BSON type.
```
Comment (MongoCaleb) [inline source/schemas/enforce-a-schema.txt:61]: This note feels like the lede for the section. Would you consider the following re-arrangment?

- Default behavior is to accept only a single type of field; "null" is a defined BSON type
- Schema fields are not nullable by default, so schema validation fails on nullable values.
- However, you can configure Realm to ignore null values in optional field types.

**Review by cbush — APPROVED** [review]

File: source/schemas/enforce-a-schema.txt [hunk 3]
```diff
@@ -43,3 +43,28 @@ Procedure
...
+   The default schema behavior is to accept only a single type for a field.
+
+
+To enable null type schema validation: 
+
+#. In the left navigation menu, select :guilabel:`App Settings` below :guilabel:`Manage`.
```
Comment (MongoCaleb) [inline source/schemas/enforce-a-schema.txt:67]: I generally try to put UI steps in the order that the user clicks. So this first step might be:

> In the left navigation menu, under Manage, select App Settings.

**Review by MongoCaleb — COMMENTED** [review]
Suggestion to reorganize a bit to put the problem statement at the beginning of the section

File: source/schemas/enforce-a-schema.txt (hunk 3 above)
Comment (mongodben) [reply source/schemas/enforce-a-schema.txt:67]: they don't actually click 'manage' which is why it's like this. manage is just a UI label

File: source/schemas/enforce-a-schema.txt (hunk 2 above)
Comment (mongodben) [reply source/schemas/enforce-a-schema.txt:61]: good point. will refactor.

File: source/schemas/enforce-a-schema.txt (hunk 3 above)
Comment (MongoCaleb) [reply source/schemas/enforce-a-schema.txt]: I still think it's useful to describe the location before the action. "In the backyard, find the $20 bill....under the rock" is less clear (to me) as "In the backyard, under the rock, find the $20 bill." I'm willing to let it go, though, if you disagree.

Comment (mongodben) [reply source/schemas/enforce-a-schema.txt]: i'm not willing to disagree. implemented your suggested changes :)

File: source/schemas/enforce-a-schema.txt [hunk 4]
```diff
@@ -43,3 +43,28 @@ Procedure
...
+
+Validate Null Types
+-------------------
+
+{+service+}'s default behavior is to only accept a single type for each field. 
+Schema fields are not nullable by default because ``null`` is a unique 
+:manual:`BSON type </reference/bson-types/>`.
+
+You can configure fields to pass schema validation for ``null`` values
+when their types are set as optional in the schema.
```
Comment (MongoCaleb) [inline source/schemas/enforce-a-schema.txt:57]: This _could_ be read as: "you can configure a field to pass schema validation if you set its type to optional." Perhaps:
You can configure any **optional** field to pass schema validation for ``null`` values.

**Review by MongoCaleb — APPROVED** [review]
LGTM with one nit

File: source/schemas/enforce-a-schema.txt (hunk 4 above)
Comment (mongodben) [reply source/schemas/enforce-a-schema.txt:57]: nice, reduces wordiness too

**Pull request merged**